2_process.R

```diff
@@ -38,5 +38,15 @@ p2_targets_list <- list(
      {
        sites_tbl <- read_csv(p2_site_list_csv)
        get_site_flowlines(p1_reaches_sf, sites_tbl, sites_crs = 4269, max_matches = 1, search_radius = 0.1)
-     })
+     }),
+  
+  # Aggregate instantaneous DO data to daily min/mean/max
+  tar_target(
+    p2_inst_data_daily,
+    aggregate_data_to_daily(p1_inst_data,p1_daily_data,"America/New_York")),
+  
+  # Combine 1) daily DO data and 2) instantaneous DO data that has been aggregated to daily 
+  tar_target(
+    p2_daily_combined,
+    bind_rows(p1_daily_data, p2_inst_data_daily))
```

Comment:
`bind_rows` will match shared column names, but in this case the data frames `p1_daily_data` and `p2_inst_data_daily` don't share the same structure, so it just fills in NA's when variables aren't available, including Value_Min for all daily sites:

![same_input](https://user-images.githubusercontent.com/8785034/146610547-427fc80d-b9c3-4962-b298-78df15faa5cf.png)

Adding 'agency_cd' and 'Parameter' to the `group_by` call within `aggregate_data_to_daily()` would save these columns in the summarized `p2_inst_data_daily` data frame. We could also add a `select` step to the filtering/aggregating pipe to omit `na_count` and `value_count` since we don't have those variables for the "dv" data. 

Do we want to retain the data quality codes? If so, we can create those within the mungeing step. If not, we should be able to use `bind_rows(p1_daily_data[,-c(Value_cd,Value_Max_cd)],p2_inst_data_daily)`

Reply:
I've aligned all of the columns except `the Value_cd` and `Value_Max_cd`. Is it a problem to have `NA` for the `Value_cd` and `Value_Max_cd`? I think I'd prefer to leave those than take out the information that is in those columns for the `dv` sites.

If we feel like we need a code, we'd have to decide what to do if there are multiple values per day. If all of the codes for one day are `A` except one which is `P` - what is the single code for that day?

Reply:
> I've aligned all of the columns except the Value_cd and Value_Max_cd in e1f8ac2.

Great, thanks!

> Is it a problem to have NA for the Value_cd and Value_Max_cd? I think I'd prefer to leave those than take out the information that is in those columns for the dv sites.

I think it depends on whether we would potentially want this information in a downstream/modeling step. Do you think there's a use case for that?

In terms of assigning a code, I wonder if it comes up often that there are multiple code values in a single day - from a QA/QC perspective I would guess that the data would get approved in whole-day increments, but I see your point! If we want a code, I wouldn't worry too much about multiple values per day and would probably just take the first one.

Reply:
Could be helpful to for diagnostics of model performance (e.g., do days with mostly P values have worse performance than days with mostly A?). A daily-aggregated "code" could be the proportion of samples that are not A.

Reply:
I agree that this could be useful. 

I'm looking at the data and there are 1231 site-days out of 100894 (0.1%) that have  multiple value codes. So if we take the "first" one for those days, we are losing some information. That includes 1107 site-days with `NaN` as the Value_cd, so there's a chance that for those days, we'd end up with a code of `NaN` (if we are taking the first and the first code is `NaN`). This would not be helpful.

Reply:
Given that it's only 0.1% of the inst. daily summaries that this applies to, I'm in favor of just being aware of this and then moving forward. I ran the code with taking just the first code and these were the numbers:
```R
> summary(as.factor(p2_inst_data_daily$Value_cd))
    A A [0]     P 
 3333     1  2288 
```
So there aren't any `NaN` that show up here.

Reply:
Another thing that comes up here is the `Value_Max_cd`. I think the "A" job is to get the `Value_Inst_cd` at the max daily value for this. Since that is basically the same functionality as what we'll need for calculating the "time at max" and that will be addressed in #34, I'd be in favor of leaving it to that issue/PR.

Reply:
Otherwise for a stand-in, we could just take the first again, knowing that in 99.9% of the site-days that is the same throughout the day.

Reply:
Thanks for checking that, Jeff. Sounds like a good approach to handle the `Value_Max_cd` with the issue you cited

Reply:
Thanks for looking into that more closely. It makes sense that in aggregating to daily time steps we would be losing some information, retaining the first value was just my initial thought on a way to summarize the quality of the uv data . Perhaps taking the last value is even more conservative, since timestamps later in the day are more likely to be tagged as "P" compared to the first value, but it probably isn't too important, since the difference spans such a small proportion of the inst. daily summaries. 

I'm curious, though - I don't see any `NaN` values for `Value_Inst_cd` in `p1_inst_data.` Were you referring to `NA` values, or are there rows where the quality code == `NaN`? If so, it seems like something wonky might be happening with the summarization.

Reply:
When the `Value_Inst` is `NA` the `Value_Inst_cd` also shows ups as `NA`:
![image](https://user-images.githubusercontent.com/6943441/146820250-9a800522-a884-424b-bfe6-f88073d7522c.png)

Reply:
OK, yep that makes sense! I thought that `NaN` values were being returned rather than just `NA`